dev/src/builtins.cpp

```diff
@@ -375,7 +375,7 @@ nfr("substring", "s,start,size", "SII", "S",
         iint size = e.ival();
         iint start = s.ival();
         if (size < 0) size = l.sval()->len - start;
-        if (start < 0 || start + size > l.sval()->len)
```

Comment:
would be good to rebase so these other changes are not in here

Reply:
rebased